### PR TITLE
Explicit add the timeout thread to default ThreadGroup

### PR DESCRIPTION
**Pull request by larskanis**
Otherwise the timeout thread would be added to the `ThreadGroup` of the thread that makes the first call to `Timeout.timeout` .

Fixes bug 19020: https://bugs.ruby-lang.org/issues/19020

Do you wish a test for the `ThreadGroup` or is it an implementation detail?

### CODE REVIEW

**Comment by eregon** [comment]
Yes, a test would be very useful to ensure there is no regression.

**Comment by larskanis** [comment]
I added a test case.

File: test/test_timeout.rb [hunk 1]
```diff
@@ -159,4 +159,24 @@ def test_fork
     assert_equal 'timeout', r.read
     r.close
   end
+
+  def test_threadgroup
+    omit 'fork not supported' unless Process.respond_to?(:fork)
```
Comment (eregon) [inline test/test_timeout.rb:164]: `assert_separately` is better so it's also tested on non-fork platforms, and the subprocess can use `assert_equal` etc

File: test/test_timeout.rb [hunk 2]
```diff
@@ -159,4 +159,24 @@ def test_fork
...
+    r, w = IO.pipe
+    # fork to make sure the timeout thread is not running
+    fork do
+      r.close
+      tg = ThreadGroup.new
+      thr = Thread.new do
+        Timeout.timeout(10){}
+      end
+      tg.add(thr)
```
Comment (eregon) [inline test/test_timeout.rb:173]: This seems racy, it should be moved inside the thread body

File: test/test_timeout.rb (hunk 1 above)
Comment (larskanis) [reply test/test_timeout.rb]: `fork` is much faster than `assert_separately`, so I thought to not slow down the tests for some corner case. Nevertheless I changed it.

**Review by eregon — APPROVED** [review]
Thank you for the fix!

**Comment by eregon** [comment]
@hsbt How do we make sure this change is propagated to ruby/ruby before the 3.2.0 release (and ideally before the next preview too) ?

**Comment by eregon** [comment]
Oh wow it's done automatically, awesome! https://github.com/ruby/ruby/commit/9d56d9975d867c94ab2a6d76e4482112ab6c3319
